packages/nimble-components/src/select/index.ts

```diff
@@ -1008,9 +1013,29 @@ export class Select
             ? this.positionAttribute
             : this.position;
 
-        this.maxHeight = this.position === SelectPosition.above
+        const availableHeight = this.position === SelectPosition.above
             ? Math.trunc(currentBox.top)
             : Math.trunc(availableBottom);
+
+        const controlHeight = parseInt(
+            controlHeightToken.getValueFor(this),
+            10
+        );
+        const listboxInnerPadding = parseInt(
+            smallPadding.getValueFor(this),
+            10
+        );
+        const listboxGap = listboxInnerPadding; // both use smallPadding
+        const listboxBorderHeight = 2; // 1px top and bottom
+        const filterHeight = this.filterMode !== FilterMode.none ? controlHeight : 0;
+        this.maxHeight = Math.min(
```

Comment:
It feels like we could single-source all of this into a static helper function that takes the element to pull the tokens from as an argument.

Reply:
I'm not 100% sure which part you think should go in a helper, but I made one that gets the dropdown height for 10.5 options. I put this in a new `helpers.ts` file under `patterns/dropdown`.

Reply:
You could take a bit further, where you have `Select` implement `DropdownPattern` (which it should be doing anyway), and then move the `avaiableHeight` calculation into the static helper method,  making its `element` parameter typed to `DropdownPattern` as well (to access the `position`).